solidity/contracts/Mailbox.sol

```diff
@@ -41,6 +44,8 @@ contract Mailbox is
     IInterchainSecurityModule public defaultIsm;
     // An incremental merkle tree used to store outbound message IDs.
     MerkleLib.Tree public tree;
+    // Hook if any post dispatch action is required
+    IMessageHook public hook;
```

Comment:
Not every user will want the same hook. Some may want none, some may want several. They should be able to choose